### PR TITLE
fix: Upgrade aurora engine version to 11.16

### PR DESCRIPTION
**Pull request by kimengu-david**
### Feature or Bugfix

- Bugfix

### Detail
Update Aurora engine version to 11.16. Fixes an issue where the Aurora nested stack deployment in the data.all backend which goes  to deployment account would fail  as AuroraPostgresEngineVersion.VER_10_18  is not compatible with parameter group default.aurora-postgresql11
### Relates
- https://github.com/awslabs/aws-dataall/pull/466

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by dlpzx** [comment]
Hi @kimengu-david thanks for opening the PR! I tested the upgrade to 11.15 on a pre-existing deployment and it failed with the following:
image

Which is a little strange given that the [documentation](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_UpgradeDBInstance.PostgreSQL.html#USER_UpgradeDBInstance.PostgreSQL.MajorVersion) states that we can upgrade from 10.18 to 11.15. In any case, I tested upgrading to 11.16 and it worked fine, but I think that we still need to understand this behavior a little better.

I have a deployment of data.all V1.4  deployed a couple of months ago that has never been updated. RDS automatically performed the upgrade from engine 10.18 to 11.16 on May 15th. The same has happened with other deployments that initially set the engine version to 10.18.

In conclusion, my guess is that the PR #466 solved a particular issue in which the upgrade must have gotten stuck. We can either:
- revert the changes in that PR and deploy with engine in major version 10 - if deployments are automatically upgraded to version 11 there must be a good reason, so staying in version 10 does not seem like the best idea
- upgrade to major version 11 (using 11.16) - we need to be careful to not get into backwards compatibility issues as the one in the screenshot

What are your thoughts?

**Review by dlpzx — CHANGES_REQUESTED** [review]
Change to ver_11_16

**Comment by kimengu-david** [comment]
I would also go for the second option. Okay updating that now.

**Review by dlpzx — APPROVED** [review]
